server/src/api/comment.rs

```diff
@@ -151,7 +151,10 @@ impl Perform for CreateComment {
 
     // Check for a community ban
     let post_id = data.post_id;
-    let post = blocking(context.pool(), move |conn| Post::read(conn, post_id)).await??;
+    let post = match blocking(context.pool(), move |conn| Post::read(conn, post_id)).await? {
+      Ok(post) => post,
+      Err(_e) => return Err(APIError::err("couldnt_find_post").into()),
+    };
```

Comment:
I would move this into a function get_post(), and also log the error (possibly at debug level).